src/main/java/com/adobe/target/edge/client/service/TelemetryService.java

```diff
@@ -91,4 +103,57 @@ private DecisioningMethod getDecisioningMethod(TargetDeliveryRequest request) {
 
     return DecisioningMethod.SERVER_SIDE;
   }
+
+  private int executeMboxCount(TargetDeliveryRequest request) {
+    int executeMboxCount = 0;
+    ExecuteRequest executeRequest = request.getDeliveryRequest().getExecute();
+    if (executeRequest != null) {
+      executeMboxCount = executeRequest.getMboxes().size();
+    }
+    return executeMboxCount;
+  }
+
+  private Boolean isExecutePageLoad(TargetDeliveryRequest request) {
+
+    boolean isExecutePageLoad = false;
+    ExecuteRequest executeRequest = request.getDeliveryRequest().getExecute();
+    if (executeRequest != null) {
+      isExecutePageLoad = executeRequest.getPageLoad() != null;
+    }
+    return isExecutePageLoad;
+  }
+
+  private Integer PrefetchMboxCount(TargetDeliveryRequest request) {
+    int prefetchMboxCount = 0;
+    PrefetchRequest prefetchRequest = request.getDeliveryRequest().getPrefetch();
+    if (prefetchRequest != null) {
+      prefetchMboxCount = prefetchRequest.getMboxes().size();
+    }
+    return prefetchMboxCount;
+  }
+
+  private Boolean isPrefetchPageLoad(TargetDeliveryRequest request) {
+    boolean prefetchPageLoad = false;
+    PrefetchRequest prefetchRequest = request.getDeliveryRequest().getPrefetch();
+    if (prefetchRequest != null) {
+      prefetchPageLoad = prefetchRequest.getPageLoad() != null;
+    }
+    return prefetchPageLoad;
+  }
+
+  private Integer prefetchViewCount(TargetDeliveryRequest request) {
+    int prefetchViewCount = 0;
+    if (request.getDeliveryRequest().getPrefetch() != null) {
+      prefetchViewCount = request.getDeliveryRequest().getPrefetch().getViews().size();
+    }
+    return prefetchViewCount;
+  }
+
+  private ExecutionMode getMode(TargetDeliveryRequest request) {
+    if (getDecisioningMethod(request).equals(DecisioningMethod.ON_DEVICE)
```

Comment:
This logic for calculating the execution mode has a caveat - we actually need to check the response status to determine where the request was actually fulfilled.  If status == 200 && (DecisioningMethod == HYBRID || DecisioningMethod == ON_DEVICE) then LOCAL

If the status != 200 it means the request couldn't be fulfilled locally and had to be routed to Delivery API.
https://github.com/adobe/target-java-sdk/blob/ee646c170e08b5e2f5e9cf8c4de5c1e441fe9ff2/src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningService.java#L244